src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -727,4 +727,15 @@ public static void AddWebAuthn(this IServiceCollection services, GlobalSettings
             });
         });
     }
+
+    public static void AddOptionality(this IServiceCollection services,
+        GlobalSettings globalSettings)
+    {
+        services.AddSingleton<ILdClient>(_ =>
+        {
+            return new LdClient(LaunchDarklyFeatureService.GetConfiguredClient(globalSettings));
```

Comment:
I would get `GlobalSettings` out of the `IServiceProvider` that is currently discarded. This gets rid of a closure and makes the `AddOptionality` method not need to supply it.

Reply:
Nice, I like it. There's a bunch of these extension methods not doing that today.